com.unity.render-pipelines.universal/Runtime/LightCookieManager.cs

```diff
@@ -534,119 +665,228 @@ int FilterAndValidateAdditionalLights(ref LightData lightData, LightCookieMappin
                 LightCookieMapping lp;
                 lp.visibleLightIndex = (ushort)i;
                 lp.lightBufferIndex  = (ushort)(i + lightBufferOffset);
+                lp.legacyLight = light;
 
                 validLightMappings[validLightCount++] = lp;
             }
 
             return validLightCount;
         }
 
-        int UpdateAdditionalLightsAtlas(CommandBuffer cmd, ref LightData lightData, ref WorkSlice<LightCookieMapping> validLightMappings, Vector4[] textureAtlasUVRects)
+        int UpdateAdditionalLightsAtlas(CommandBuffer cmd, ref WorkSlice<LightCookieMapping> validLightMappings, Vector4[] textureAtlasUVRects)
```

Comment:
It would be great to have test coverage of this. I feel like this would be really easy to break if the code is touched for a bug fix for example.

Reply:
Yes, it became more complex than I initially thought. It definitely needs a test.
I'm planning to add more tests (in general) in another PR.